### PR TITLE
[resign.sh] Fix icloud-container-environment value

### PR DESCRIPTION
**Pull request by ryanprobus**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

The `com.apple.developer.icloud-container-environment` entitlement was being incorrectly set to `development` when using app entitlements if passing in a SHA1 as the `CERTIFICATE` parameter. This behavior appears to have been introduced by this [pull request](https://github.com/fastlane/fastlane/pull/4898/files) which added the ability to pass the SHA1 instead of the certificate name.

Partially fixes this [issue](https://github.com/fastlane/fastlane/issues/13740) pertaining to entitlements when resigning.

### Description

When iterating over the entitlement transfer rules, there is a specific check for `com.apple.developer.icloud-container-environment`, within this check, I added a regex to check if `CERTIFICATE` matches a SHA1. If it does, I get the name of the identity by grepping `security find-identity`.

Used the below script to test various types of inputs including non-SHA1 inputs. Additionally, I resigned our IPAs using the fastlane `resign` action and checked the embedded entitlements. The entitlement was correctly set to `production` if the certificate's name contained `Distribution:`.

```
CERTIFICATE="CERTIFICATE_VALUE"
sha1_pattern='[0-9A-F]{40}'
if [[ "$CERTIFICATE" =~ $sha1_pattern ]]; then
  certificate_matches="$(security find-identity -v -p codesigning | grep -m 1 "$CERTIFICATE")"
  if [ -n "$certificate_matches" ]; then
    certificate_name="$( sed -E s/[^\"]+\"\([^\"]+\)\".*/\\1/ <<< $certificate_matches )"
    echo "$certificate_name"
  fi
fi
```

### CODE REVIEW

File: sigh/lib/assets/resign.sh [hunk 1]
```diff
@@ -743,7 +743,19 @@ function resign {
                 # This value is set by Xcode during export (manually selected for Development and AdHoc, automatically set to Production for Store)
                 # Would need an additional dedicated option to specify the iCloud environment to be used (Development or Production)
                 # For now, we assume Production is to be used when signing with a Distribution certificate, Development if not
-                if [[ "$CERTIFICATE" =~ "Distribution:" ]]; then
+                local certificate_name=$CERTIFICATE
+                local sha1_pattern='[0-9A-F]{5,40}'
```
Comment (lyndsey-ferguson) [inline sigh/lib/assets/resign.sh]: this should be a straight 40

**Review by lyndsey-ferguson — CHANGES_REQUESTED** [review]
one change requested

File: sigh/lib/assets/resign.sh [hunk 2]
```diff
@@ -743,7 +743,19 @@ function resign {
...
+
+                if [[ "$CERTIFICATE" =~ $sha1_pattern ]]; then
+                    log "Certificate $CERTIFICATE matches a SHA1 pattern"
+                    local certificate_matches="$( security find-identity -v -p codesigning | grep "$CERTIFICATE" )"
```
Comment (kpink224) [inline sigh/lib/assets/resign.sh]: Let's do `grep -m 1 "$CERTIFICATE"` to ensure we only get one option. It will cause problems if we get multiple.

**Review by lyndsey-ferguson — APPROVED** [review]
LGTM

File: sigh/lib/assets/resign.sh [hunk 3]
```diff
@@ -743,7 +743,19 @@ function resign {
...
                 # For now, we assume Production is to be used when signing with a Distribution certificate, Development if not
-                if [[ "$CERTIFICATE" =~ "Distribution:" ]]; then
+                local certificate_name=$CERTIFICATE
+                local sha1_pattern='[0-9A-F]{40}'
+
+                if [[ "$CERTIFICATE" =~ $sha1_pattern ]]; then
+                    log "Certificate $CERTIFICATE matches a SHA1 pattern"
+                    local certificate_matches="$( security find-identity -v -p codesigning | grep "$CERTIFICATE" )"
+                    if [ -n "$certificate_matches" ]; then
+                        certificate_name="$( sed -E s/[^\"]+\"\([^\"]+\)\".*/\\1/g <<< $certificate_matches )"
```
Comment (kpink224) [inline sigh/lib/assets/resign.sh]: you don't need `g` at the end of sed

**Review by kpink224 — COMMENTED** [review]
LG just some minor changes

**Review by kpink224 — APPROVED** [review]

**Review by joshdholtz — CHANGES_REQUESTED** [review]
@ryanprobus Sorry about just getting to this! I will admin that I am not a `resign.sh` expert but this looks like a solid change so 🙌🙌 to you for making this 😊

One request I have is if you could add to the change log at the top of the file since that is kind of what has been happening - https://github.com/fastlane/fastlane/blob/b188b9d9c9a3a739e875312a58fb2c008b23cff6/sigh/lib/assets/resign.sh#L42-L66

Thanks!

**Comment by ryanprobus** [comment]
@joshdholtz No problem, thanks for taking a look at this. A lot of people are afraid of `resigh.sh`, but its definitely something could use some loving. I added a note to the change log and squashed the commit.

**Review by joshdholtz — APPROVED** [review]
@ryanprobus Thanks for adding that note and giving `resign.sh` the love that it needs 😊 You are a true hero 🙌

**Pull request merged**